### PR TITLE
fixed meteosat native georeferencing

### PR DESCRIPTION
**Pull request by hazbottles**
- Fixes georeferencing bug with meteosat .nat files. Results demonstrated here over several different satellites (note the middle and right images have `TypeOfEarthModel=1`, left image is post Dec 2017 and has `TypeOfEarthModel=2`). Right = MSG-4, middle=MSG-3, left=MSG-2. 
image

 - [x] Tests added 
 - [x] Tests passed 
 - [X] Passes ``git diff origin/master **/*py | flake8 --diff``

### CODE REVIEW

File: satpy/readers/native_msg.py [hunk 1]
```diff
@@ -283,11 +295,25 @@ def get_area_extent(self, dsid):
                 "ReferenceGridVIS_IR"]["ColumnDirGridStep"] * 1000.0
             line_step = data15hd['ImageDescription'][
                 "ReferenceGridVIS_IR"]["LineDirGridStep"] * 1000.0
-
-            ll_c = (center_point - west - 0.5) * column_step
-            ll_l = (south - center_point + 1.5) * line_step
-            ur_c = (center_point - east + 0.5) * column_step
-            ur_l = (north - center_point + 0.5) * line_step
+            # section 3.1.4.2 of MSG Level 1.5 Image Data Format Description
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/readers/native_msg.py]: So this fix should be dependent on the date of the data right ? I think we do that in the hrit reader for seviri

File: satpy/readers/native_msg.py [hunk 2]
```diff
@@ -283,11 +295,25 @@ def get_area_extent(self, dsid):
...
+                ns_offset = 0.5  # north +ve
+                we_offset = 0.5  # west +ve
+            else:
+                raise NotImplementedError(
+                    'unrecognised earth model: {}'.format(earth_model)
+                )
+
+            # section 3.1.5 of MSG Level 1.5 Image Data Format Description
+            ll_c = (center_point - west - 0.5 + we_offset) * column_step
+            ll_l = (south - center_point - 0.5 + ns_offset) * line_step
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/readers/native_msg.py]: have you checked with multiple scenes ? the satellite has a tendency to wobble...

File: satpy/readers/native_msg.py [hunk 3]
```diff
@@ -272,6 +273,17 @@ def get_area_extent(self, dsid):
 
         if dsid.name != 'HRV':
 
+            # following calculations assume grid origin is south-east corner
```
[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/readers/native_msg.py]: thanks for making this clear

**Comment by mraspaud** [comment]
@hazbottles any answers to my questions ?

File: satpy/readers/native_msg.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hazbottles) [reply satpy/readers/native_msg.py]: Sorry, away from my computer till Monday, will check some other scenes. 

Note also that the new southern border is in line with my interpretation of the documentation (link in module doc string)

File: satpy/readers/native_msg.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hazbottles) [reply satpy/readers/native_msg.py]: The TypeOfEarthModel field in the .nat file is enough to know whether to include the 0.5 pixel offset, so don’t need to worry about date based logic. I’ll have a look at the HRIT reader when I’m in front of a machine Monday.

[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/readers/native_msg.py]: sounds good, have a nice week-end

File: satpy/readers/native_msg.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hazbottles) [reply satpy/readers/native_msg.py]: Updated the description with some images of other scenes. Yes, this does appear to be a universal improvement.

File: satpy/readers/native_msg.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hazbottles) [reply satpy/readers/native_msg.py]: Glad you asked me to look at some other scenes, seems I had the n/s offset the wrong way!. Fixed now (and images in first description comment are up-to-date with code)

**Comment by mraspaud** [comment]
@ColinDuff @sjoro any comments on this pr ?

**Comment by mraspaud** [comment]
@hazbottles thanks for you contribution. Do you think you could also add a small unittest to make sure we don't screw up you work on geolocation in the future ?

**Comment by hazbottles** [comment]
sure, I'll add a test or two tomorrow :)

**Comment by hazbottles** [comment]
Some unit tests added.

**Pull request merged**